Marke45/lablog
# Assay refactor

I've refactored some of the code around the assay generation and assignent.  The 'get_assay' code is now in a 'service layer' (a place where we can put business logic away from view code and model code) which, will help us keep the code separated, reduce repetition and therefore bugs, and make testing so much easier!

To evidence the testing I ahve added a test for the get_assay function.

I have alos changed the liquid standard model slightly.  I have changed the type to a JSONField.  This type enables the easy storing, serialisation and deserialisation of data to python objects.  This seems like a good thing for what we are trying to do (store a simple list that doesn't really require a model).  This has made the logic for the get_assay fucntion a lot simpler too!

Hope you like.

**To test**

Pull branch & load virtual env
run pipenv install - to install pytest
run python manage.py migrate - to migrate the JSONField change
run pytest - will run all tests and should pass
use app as normal and it should work as normal.

Note:  You'll notice a slightly different UI item in the edit view of liquid standard for the JSONField.  We can customise this to be nicer later.
